File: atcoder/practice/ABC229/C.py
n , w = (int(x) for x in input().split())

cheese = [[0]*2 for _ in range(n)]

for i in range(n):
    cheese[i] = list(map(int, input().split()))

cheese.sort()
cheese.reverse()


ans = 0
for x in cheese:
    ans += x[0] * min(x[1], w)
    w -= min(x[1], w)

print(ans)

File: atcoder/practice/ABC229/C.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
  long long n,w;
  cin >> n >> w;
  vector<pair<long long,long long>> v(n);
  for(auto &nx : v){
    cin >> nx.first >> nx.second;
  }
  cout << endl;
  sort(v.begin(),v.end());
  for(auto &nx : v){
      cout << nx.first  << " "<< nx.second << endl;
  }
  cout << '\n';
  reverse(v.begin(),v.end());

  for(auto &nx : v){
      cout << nx.first  << " "<< nx.second << endl;
  }
  cout << '\n';
  long long res=0;
  for(auto &nx : v){
    res+=nx.first*min(w,nx.second);
    w-=min(w,nx.second);
  }
  cout << res << '\n';
  return 0;
}